src/main/java/com/wanderersoftherift/wotr/abilities/attachment/AttachedEffectData.java

```diff
@@ -180,16 +178,18 @@ public AttachedEffect(AttachEffect effect, AbilityContext context) {
          * Ticks this attached effect
          * 
          * @param attachedTo The entity it is attached to
-         * @param level      The level it is within
          * @return Whether this effect has expired and should be removed
          */
-        public boolean tick(Entity attachedTo, ServerLevel level) {
+        public boolean tick(Entity attachedTo) {
+            if (!(attachedTo.level() instanceof ServerLevel level)) {
+                return true;
+            }
             LivingEntity caster = getCaster(level);
```

Comment:
Caster might not be in the same level as the entity.
```suggestion
            LivingEntity caster = getCaster(level.getServer());
```